sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -556,14 +738,74 @@ export class EventHubSender extends LinkEntity {
     }
   }
 
+  private _transformEvents(
```

Comment:
I'd love to see unit tests around _just_ this function. It seems complicated enough that it'd be worth it.

Reply:
I extracted this method into its own function and added tests around it. Doing so I discovered I was actually setting the PENDING_PUBLISH_SEQ_NUM_SYMBOL on the event even if idempotency wasn't enabled! (Which wouldn't really have much impact since I used a symbol, but it's something!)